bundler/spec/dependabot/bundler/update_checker/force_updater_spec.rb

```diff
@@ -306,5 +306,24 @@
         )
       end
     end
+
+    context "when peer dependencies in the Gemfile shouldn't update together, because one of them would be dowgraded" do
+      let(:dependency_files) { bundler_project_dependency_files("no_downgrades") }
+      let(:target_version) { "7.1.1" }
+      let(:dependency_name) { "rails" }
+      let(:requirements) do
+        [{
+          file: "Gemfile",
+          requirement: "~> 7.1",
+          groups: [:default],
+          source: nil
+        }]
+      end
+
+      it "raises a resolvability error" do
+        expect { updater.updated_dependencies }
+          .to raise_error(Dependabot::DependencyFileNotResolvable)
```

Comment:
Wondering if it makes sense to _raise_ in this scenario, wouldn't it be better to simply not perform the update and say we're at the latest possible version? I thought that's what we do in most of these sort of scenario's?

Reply:
Yep, that's how it works.

I assert for that error since I'm unit testing the force updater, but that error is gracefully handled here:

https://github.com/dependabot/dependabot-core/blob/b39f6b0eeecf8140943e924e93838f65835f7572/bundler/lib/dependabot/bundler/update_checker.rb#L150-L170

Reply:
Ah thanks for clarifying!